yugabyte/src/yugabyte/ysql/append.clj

```diff
@@ -36,11 +36,24 @@
   [test k]
   (str "v" (mod k keys-per-row)))
 
+(defn select-with-pessimistic
+  [locking col table]
+  (let [clause (if (= :pessimistic locking)
+                 (rand-nth ["" " for update" " for no key update" " for share" " for key share"])
+                 "")]
+    (str "select (" col ") from " table " where k = ?" clause)))
+
+(defn lock-row-if-needed
+  [locking conn col table v row]
+  (if (= :pessimistic locking)
+    (c/execute! conn [(select-with-pessimistic locking col table) v row]))
```

Comment:
why do we need to do a select with lock before each update if we're testing wait-on-conflict concurrency control?